### PR TITLE
[pilot] Workaround for glog Parsing Error

### LINKED ISSUE #2127: ERROR: logging before flag.Parse
This message shows up in the Pilot logs:
`ERROR: logging before flag.Parse: <log msg>`

I wonder if this was accidentally introduced during some of the recent refactoring of the cmds. @nmittler?

### PR DESCRIPTION
**Pull request by nmittler**
This is copied from mixer code (introduced by https://github.com/istio/istio/commit/5ae17d07ead24e31fba01c91673122bba2b6e804)

Fixes #2127

**Release note**:
```release-note
None
```

### CODE REVIEW

File: pilot/cmd/pilot-discovery/main.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"github.com/spf13/cobra"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"flag"
```
Comment (ZackButcher) [inline pilot/cmd/pilot-discovery/main.go]: run goimport please, this should be up with the other golang standard libraries.

Comment (nmittler) [reply pilot/cmd/pilot-discovery/main.go]: done

**Comment by nmittler** [comment]
@ZackButcher PTAL

**Pull request merged**